src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/SelectVariants.java

```diff
@@ -0,0 +1,1244 @@
+    private Set<String> IDsToRemove = null;
+
+    private final List<Allele> diploidNoCallAlleles = Arrays.asList(Allele.NO_CALL, Allele.NO_CALL);
+
+    /**
+     * Set up the VCF writer, the sample expressions and regexs, filters inputs, and the JEXL matcher
+     *
+     */
+    @Override
+    public void onTraversalStart() {
+        final FeatureInput<? extends Feature> featureInput = getDrivingVariantsFeatureInput();
+
+        final Map<String, VCFHeader> vcfRods = new HashMap<>();
+        final VCFHeader vcfHeader= getHeaderForVariants();
+        vcfRods.put(featureInput.getName(), vcfHeader);
+
+        // Initialize VCF header lines
+        final Set<VCFHeaderLine> headerLines = createVCFHeaderLineList(vcfRods);
+
+        for (int i = 0; i < selectExpressions.size(); i++) {
+            // It's not necessary that the user supply select names for the JEXL expressions, since those
+            // expressions will only be needed for omitting records.  Make up the select names here.
+            selectNames.add(String.format("select-%d", i));
+        }
+
+        jexls = VariantContextUtils.initializeMatchExps(selectNames, selectExpressions);
+
+        // Prepare the sample names and types to be used by the corresponding filters
+        samples = createSampleNameInclusionList(vcfRods);
+        selectedTypes = createSampleTypeInclusionList();
+
+        // Look at the parameters to decide which analysis to perform
+        discordanceOnly = discordanceTrack != null;
+        if (discordanceOnly) {
+            logger.info("Selecting only variants discordant with the track: " + discordanceTrack.getName());
+        }
+
+        concordanceOnly = concordanceTrack != null;
+        if (concordanceOnly) {
+            logger.info("Selecting only variants concordant with the track: " + concordanceTrack.getName());
+        }
+
+        if (mendelianViolations) {
+            sampleDB = initializeSampleDB();
+            mv = new MendelianViolation(mendelianViolationQualThreshold, false, true);
+        }
+
+        selectRandomFraction = fractionRandom > 0;
+        if (selectRandomFraction) {
+            logger.info("Selecting approximately " + 100.0*fractionRandom + "% of the variants at random from the variant track");
+        }
+
+        // Prepare the variant IDs to keep and to be removed for use by the IDs filter
+        IDsToKeep = getIDsFromFile(rsIDFile);
+        IDsToRemove = getIDsFromFile(XLrsIDFile);
+
+        VariantContextWriterBuilder vcWriterBuilder =
+                new VariantContextWriterBuilder()
+                    .setOutputFile(outFile)
+                    .setOutputFileType(VariantContextWriterBuilder.OutputType.VCF)
+                    .unsetOption(Options.INDEX_ON_THE_FLY);
+        if (lenientVCFProcesssing) {
+            vcWriterBuilder = vcWriterBuilder.setOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER);
+        }
+        vcfWriter = vcWriterBuilder.build();
```

Comment:
Can you extract a utility method `createVCFWriter(File outFile, boolean indexOnTheFly, Options... additionalOptions)`, put it in `GATKVariantContextUtils`, and call it from here?